make/tests

```diff
@@ -29,9 +29,12 @@ test/%$(EXE) : test/%.o $(GTEST)/src/gtest_main.cc $(GTEST)/src/gtest-all.o $(MP
 
 ##
 # Include dependency files for tests
+# This has the side-effect of making the object files non-intermediates,
+# so we need to explicitly tell make we don't want to keep them
 ##
 ifneq ($(filter test/%,$(MAKECMDGOALS)),)
 -include $(patsubst %$(EXE),%.d,$(filter test/%,$(MAKECMDGOALS)))
+.INTERMEDIATE: $(patsubst %$(EXE),%.o,$(filter-out test/%.d test/%.o,$(filter test/%,$(MAKECMDGOALS))))
```

Comment:
Just a suggestion or maybe a thought on how to better pattern match, how about matching the executable a little more:

```
.INTERMEDIATE: $(patsubst test/%_test$(EXE),test/%_test.o,$(filter test/%_test$(EXE),$(MAKECMDGOALS)))
```

(I didn't run it... so it might not be correct.)

what I'm thinking is that we know that we want a list of `.o` when the target looks like `test/*_test` (or `test/*_test.exe`).

If the `.o` were provided as a target directly we could leave it alone?

Reply:
This worked better, thanks!